pkg/querier/frontend/frontend.go

```diff
@@ -42,16 +44,18 @@ var (
 
 // Config for a Frontend.
 type Config struct {
-	MaxOutstandingPerTenant int    `yaml:"max_outstanding_per_tenant"`
-	CompressResponses       bool   `yaml:"compress_responses"`
-	DownstreamURL           string `yaml:"downstream"`
+	MaxOutstandingPerTenant int           `yaml:"max_outstanding_per_tenant"`
+	CompressResponses       bool          `yaml:"compress_responses"`
+	DownstreamURL           string        `yaml:"downstream"`
+	LogQueriesLongerThan    time.Duration `yaml:"log_queries_longer_than"`
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxOutstandingPerTenant, "querier.max-outstanding-requests-per-tenant", 100, "Maximum number of outstanding requests per tenant per frontend; requests beyond this error with HTTP 429.")
 	f.BoolVar(&cfg.CompressResponses, "querier.compress-http-responses", false, "Compress HTTP responses.")
 	f.StringVar(&cfg.DownstreamURL, "frontend.downstream-url", "", "URL of downstream Prometheus.")
+	f.DurationVar(&cfg.LogQueriesLongerThan, "frontend.log-queries-longer-than", 0, "Log slow queries")
```

Comment:
Maybe something like: `Log queries that are slower than the specified duration. 0 to disable.`.

Reply:
updated.